Source/SwiftLintFramework/Rules/Idiomatic/TypeNameRule.swift

```diff
@@ -68,7 +68,13 @@ public struct TypeNameRule: ASTRule, ConfigurationProviderRule {
             return []
         }
 
+        #if canImport(SwiftUI)
+        let name = name
+            .nameStrippingLeadingUnderscoreIfPrivate(dictionary)
+            .nameStrippingTrailingSwiftUIPreviewProvider(dictionary)
+        #else
         let name = name.nameStrippingLeadingUnderscoreIfPrivate(dictionary)
+        #endif
```

Comment:
The compiler directives seem to be causing the failures, which are due to `testRulesDocumentationIsUpdated` failing.  Initially there was no compiler directive but there were some significant (~35% slower) performance.  I wasn't sure if I should worry about these performance issues.  Also, anyone on Linux doesn't need this fix.  Any suggestions on how to proceed @marcelofabri, @jpsim?